### DISCUSSION TITLE
Integrate a new property into TextInput to manage the "shortcuts" visibility on iPads

### DISCUSSION THREAD

**Discussion by rezkiy37** [react-native-community/discussions-and-proposals#830] [Ideas]
I’m tackling an issue [IOU - Empty suggestion pop-up appears below numeric keyboard, preventing number input #50057](https://github.com/Expensify/App/issues/50057) at **Expensify**. Here’s a quick summary:

This issue affects the **TextInput** on **iPads** only. There is an amount input with a custom numeric keyboard on the display. When the user interacts with it a blocking view appears. We refer to it as the “**suggestion pop-up**” (or “**shortcuts**” in settings). It appears after each interaction with the input field. Although it disappears once the user taps it, the pop-up intercepts clicks on buttons, requiring users to tap twice to interact with the app.

Unfortunately, **React Native** doesn’t currently provide a built-in way to hide this suggestion pop-up programmatically.

However, **iOS** does offer a native property for **UITextField** called `inputAssistantItem`. According to the [documentation](https://developer.apple.com/documentation/uikit/uitextinputassistantitem), we can hide the **"shortcuts"** by setting the `leadingBarButtonGroups` and `trailingBarButtonGroups` properties to empty arrays.

I propose adding a new property for **TextInput** in **React Native**, which would set these native properties to empty arrays. This new property could be called `disableInputAssistant` or `disableKeyboardShortcuts` and would be a `boolean`.

<details><summary>Current behaviour</summary>

https://github.com/user-attachments/assets/d1ebbef9-856a-4cf8-999b-e7da9fd21853

</details> 

As you can see the user cannot interact with the input freely.

<details><summary>Changes</summary>

```objective-c
textField.inputAssistantItem.leadingBarButtonGroups = @[];
textField.inputAssistantItem.trailingBarButtonGroups = @[];
``` 

https://github.com/user-attachments/assets/6854a230-184b-4602-a278-8e76eac4c10c

<img width="1722" alt="Screenshot 2024-10-29 at 12 59 42" src="https://github.com/user-attachments/assets/254fd531-f3ae-4694-96de-e576e7beeced">
<img width="1728" alt="Screenshot 2024-10-29 at 13 01 03" src="https://github.com/user-attachments/assets/1f5984b7-4693-4ddb-96e9-68d70bc7078a">

</details> 

After applying these native props the user taps on the software buttons normally. Moreover, it allows us to manage the redo & undo buttons.

**Comment by rezkiy37** [comment]
Meanwhile, I'm preparing a draft PR 🙂

**Comment by mjmasn** [comment]
Would be good to see this officially supported / fixed. We hit a similar issue and patched it in as a new `assistantBarHidden` prop (to match the `caretHidden` and `contextMenuHidden` props naming-wise).

**Comment by rezkiy37** [comment]
That's my PR - https://github.com/facebook/react-native/pull/47671. I am working on it opening asap.

**Comment by rezkiy37** [comment]
The PR (https://github.com/facebook/react-native/pull/47671) has been opened for review 🙂

**Comment by rezkiy37** [comment]
https://github.com/facebook/react-native/pull/47671#issuecomment-2580627954
